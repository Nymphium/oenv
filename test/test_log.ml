open Oenv

let () =
  Logs.(
    set_reporter @@ format_reporter ();
    set_level @@ Some Info);
  let var1 = "var1" in
  let var2 = "var2" in
  let getter = function
    | x when x = var1 -> Some "hello"
    | x when x = var2 -> Some "world"
    | _ -> None
  in
  let reader =
    Product.(v (fun l r -> l, r) +: string var1 +: string ~secret:false var2 |> close)
  in
  ignore @@ read_source reader getter
;;
