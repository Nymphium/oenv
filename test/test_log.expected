test_log.exe: [INFO] Found var1: <secret>
test_log.exe: [INFO] Found var2: world
